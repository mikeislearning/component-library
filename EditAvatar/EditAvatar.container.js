import { connect } from 'react-redux';
import { createImage } from 'app/actions';

import EditAvatar from './EditAvatar';

const mapStateToProps = (state) => ({
  currentUser: state.currentUser?.data,
  isLoading: state.images.loading,
});

export default connect(
  mapStateToProps,
  { createImage }
)(EditAvatar);
