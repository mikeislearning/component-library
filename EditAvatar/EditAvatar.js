// @format
import React from 'react';
import { PropTypes } from 'prop-types';
import { Text, View, ScrollView, TouchableOpacity, Image } from 'react-native';

import { Button, Loading, UserImage, Tips } from 'app/components';
import { avatarImages, openCamera, openPicker } from 'app/utils';
import { logEvent, sendSentryError } from 'app/services';
import { images } from 'app/theme';

import styles from './EditAvatar.styles';

export default class EditAvatar extends React.Component {
  static propTypes = {
    avatar: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    currentUser: PropTypes.object.isRequired,
    createImage: PropTypes.func.isRequired,
    isLoading: PropTypes.bool,
    isOnboarding: PropTypes.bool,
    setAvatar: PropTypes.func,
  };

  static defaultProps = {
    isOnboarding: false,
  };

  constructor() {
    super();

    this.avatars = [];
    avatarImages.map((image) => {
      this.avatars.push({
        accessibilityLabel: image.accessibilityLabel,
        name: image.name,
        src: images[image.name],
      });
    });
  }

  updateImage = (type) => {
    const photoFunction = type === 'picker' ? openPicker : openCamera;
    let eventToTrack;

    if (this.props.isOnboarding) {
      eventToTrack =
        type === 'picker' ? `setup_profile_s2_upload_photo` : 'setup_profile_s2_take_photo';
    } else {
      eventToTrack = type === 'picker' ? `profile_upload_photo` : 'profile_take_photo';
    }

    logEvent(eventToTrack);

    photoFunction(async (response) => {
      if (response) {
        try {
          // TODO - change so it gets this value from global state
          const avatar = await this.props.createImage(response);
          this.props.setAvatar(avatar);
        } catch (error) {
          sendSentryError('Error creating an image file from a profile picture', error);
        }
      }
    });
  };

  render() {
    const { avatar, currentUser, isLoading, setAvatar } = this.props;
    const tipsText = 'We have noticed that people uploading a real picture get more connections. ';

    if (currentUser.loading) return <Loading />;

    return (
      <React.Fragment>
        <Tips text={tipsText} />
        <View style={styles.avatarContainer}>
          {isLoading ? (
            <Loading />
          ) : (
            <UserImage src={avatar} resizeMode={'contain'} size="massive" />
          )}
        </View>
        <View style={styles.avatarButtons}>
          <View style={[styles.button, styles.takePhotoButton]}>
            <Button
              accessibilityLabel="Take a photo with your camera"
              onPress={() => this.updateImage('camera')}
              theme="bordered"
            >
              <View style={styles.row}>
                <Image source={images.camera} style={[styles.buttonIcon, styles.cameraIcon]} />
                <Text style={styles.buttonText}>Take photo</Text>
              </View>
            </Button>
          </View>
          <View style={[styles.button, styles.uploadPhotoButton]}>
            <Button
              accessibilityLabel="Upload a photo from your gallery"
              onPress={() => this.updateImage('picker')}
              theme="bordered"
            >
              <View style={styles.row}>
                <Image source={images.upload} style={[styles.buttonIcon, styles.uploadIcon]} />
                <Text style={styles.buttonText}>Upload photo</Text>
              </View>
            </Button>
          </View>
        </View>
        <Text style={styles.title}>Or, choose an avatar </Text>
        <ScrollView style={styles.avatarImagesContainer} horizontal={true}>
          {this.avatars.map((image) => (
            <TouchableOpacity
              accessibilityLabel={image.accessibilityLabel}
              key={image.name}
              onPress={() => setAvatar(image.name)}
              style={[
                styles.avatarImageItem,
                avatar === image.name ? styles.avatarImageItemSelected : null,
              ]}
            >
              <Image source={image.src} style={styles.avatarImage} resizeMode={'contain'} />
            </TouchableOpacity>
          ))}
        </ScrollView>
      </React.Fragment>
    );
  }
}
