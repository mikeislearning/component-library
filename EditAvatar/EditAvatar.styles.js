import { StyleSheet } from 'react-native';
import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  avatarButtons: {
    alignItems: 'center',
    alignSelf: 'center',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    marginBottom: 15,
    marginHorizontal: -7,
  },
  avatarContainer: {
    alignItems: 'center',
    alignSelf: 'center',
    borderColor: colors.gray.light,
    borderRadius: 200,
    borderWidth: 1,
    height: 160,
    marginBottom: 30,
    width: 160,
  },
  avatarImage: {
    height: 60,
    margin: 10,
    width: 60,
  },
  avatarImageItem: {
    borderColor: colors.gray.light,
    borderRadius: 60,
    borderWidth: 1,
    marginRight: 10,
  },
  avatarImageItemSelected: {
    backgroundColor: colors.green.light,
    borderColor: colors.green.dark,
  },
  avatarImagesContainer: {
    display: 'flex',
    flexDirection: 'row',
    marginBottom: 20,
  },
  button: {
    alignSelf: 'center',
    marginBottom: 15,
    marginHorizontal: 7,
  },
  buttonIcon: {
    marginRight: 13,
  },
  buttonText: {
    color: colors.blue,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: 16,
  },
  cameraIcon: {
    height: 16,
    width: 19,
  },
  row: {
    alignItems: 'center',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  takePhotoButton: {
    width: 150,
  },
  title: {
    alignSelf: 'center',
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: fonts.size.h5,
    marginBottom: 20,
  },
  uploadIcon: {
    height: 19,
    width: 16,
  },
  uploadPhotoButton: {
    width: 170,
  },
});

export default styles;
