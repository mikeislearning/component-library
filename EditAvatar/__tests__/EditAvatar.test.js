import React from 'react';
import { render } from 'react-native-testing-library';

import EditAvatar from '../EditAvatar';

describe('<EditAvatar />', () => {
  const currentUser = {
    avatar: {
      uploaded: 'emptyProfile',
    },
  };

  it('renders correctly', () => {
    const editAvatar = render(
      <EditAvatar
        avatar="emptyProfile"
        createImage={() => {}}
        currentUser={currentUser}
        setAvatar={() => {}}
        updateProfilePicture={() => {}}
      />
    );
    expect(editAvatar).toMatchSnapshot();
  });
});
