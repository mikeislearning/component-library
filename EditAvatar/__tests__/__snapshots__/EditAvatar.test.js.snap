// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditAvatar /> renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 13,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "marginBottom": 3,
          "width": 30,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/images/bulb-icon.png",
          }
        }
        style={
          Object {
            "height": 30,
            "marginBottom": 5,
            "marginTop": 5,
            "width": 30,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "color": "#272E39",
          "flexDirection": "row",
          "fontFamily": "MessinaSans-RegularItalic",
          "fontSize": 15,
          "fontStyle": "italic",
          "fontWeight": "300",
          "lineHeight": 25,
          "marginLeft": 15,
          "width": "85%",
        }
      }
    >
      We have noticed that people uploading a real picture get more connections. 
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#ECECEC",
        "borderRadius": 200,
        "borderWidth": 1,
        "marginBottom": 30,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#ECECEC",
            "borderRadius": 160,
            "borderWidth": 1,
            "overflow": "hidden",
          },
          Object {
            "height": 160,
            "width": 160,
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/images/empty-profile.png",
          }
        }
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "justifyContent": "center",
            },
            Object {
              "height": 160,
              "width": 160,
            },
          ]
        }
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginBottom": 15,
        "marginHorizontal": -7,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "marginBottom": 15,
            "marginHorizontal": 7,
          },
          Object {
            "width": 150,
          },
        ]
      }
    >
      <View
        accessibilityLabel="Take a photo with your camera"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#4145C8",
            "borderRadius": 6,
            "borderWidth": 1,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../app/images/camera.png",
              }
            }
            style={
              Array [
                Object {
                  "marginRight": 13,
                },
                Object {
                  "height": 16,
                  "width": 19,
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#4145C8",
                "fontFamily": "MessinaSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Take photo
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "marginBottom": 15,
            "marginHorizontal": 7,
          },
          Object {
            "width": 170,
          },
        ]
      }
    >
      <View
        accessibilityLabel="Upload a photo from your gallery"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#4145C8",
            "borderRadius": 6,
            "borderWidth": 1,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 14,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../app/images/upload.png",
              }
            }
            style={
              Array [
                Object {
                  "marginRight": 13,
                },
                Object {
                  "height": 19,
                  "width": 16,
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#4145C8",
                "fontFamily": "MessinaSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Upload photo
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "MessinaSans-Regular",
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    Or, choose an avatar 
  </Text>,
  <RCTScrollView
    horizontal={true}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 20,
      }
    }
  >
    <View>
      <View
        accessibilityLabel="Man's face in glasses with short hair and beard"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#ECECEC",
            "borderRadius": 60,
            "borderWidth": 1,
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/images/avatars/avatar-1.png",
            }
          }
          style={
            Object {
              "height": 60,
              "margin": 10,
              "width": 60,
            }
          }
        />
      </View>
      <View
        accessibilityLabel="Woman's face with styled hair"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#ECECEC",
            "borderRadius": 60,
            "borderWidth": 1,
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/images/avatars/avatar-2.png",
            }
          }
          style={
            Object {
              "height": 60,
              "margin": 10,
              "width": 60,
            }
          }
        />
      </View>
      <View
        accessibilityLabel="Woman's face with covered head and shoulders"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#ECECEC",
            "borderRadius": 60,
            "borderWidth": 1,
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/images/avatars/avatar-3.png",
            }
          }
          style={
            Object {
              "height": 60,
              "margin": 10,
              "width": 60,
            }
          }
        />
      </View>
      <View
        accessibilityLabel="Man's face with a turban covering his head"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#ECECEC",
            "borderRadius": 60,
            "borderWidth": 1,
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/images/avatars/avatar-4.png",
            }
          }
          style={
            Object {
              "height": 60,
              "margin": 10,
              "width": 60,
            }
          }
        />
      </View>
      <View
        accessibilityLabel="Asian man's face"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#ECECEC",
            "borderRadius": 60,
            "borderWidth": 1,
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/images/avatars/avatar-5.png",
            }
          }
          style={
            Object {
              "height": 60,
              "margin": 10,
              "width": 60,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>,
]
`;
